### PR TITLE
[repo] Add RELEASENOTES and READMEs to NuGet packages

### LINKED ISSUE #4895: Add release notes and change log to NuGet
# Feature Request

**Is your feature request related to a problem?**

The current NuGet packages published from this repo and opentelemetry-dotnet-contrib repo look like 
image

**Describe the solution you'd like:**

Provide reasonable docs and change history.

image

## Additional Context

This is how .NET team has solved it https://github.com/dotnet/sdk/blob/c54cc3c8af8d8a7fb182efc080c00baf2765fc8d/src/Containers/packaging/package.csproj#L42-L47.

### PR DESCRIPTION
**Pull request by RichardChukwu**
Resolves #4895

## Changes

* Updates `Common.prod.props` so that project `README.md` files and repo `RELEASENOTES.md` are included in NuGet packages during `dotnet pack`. Relative links are automatically transformed into valid permalinks.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)

### CODE REVIEW

File: README.md
Comment (reyang) [inline README.md:181]: Remove extra blank line.

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (reyang) [inline src/OpenTelemetry/OpenTelemetry.csproj:32]: nit:

```suggestion
    <PackageReadmeFile>README.md</PackageReadmeFile>
    <PackageReleaseNotes>RELEASENOTES.md</PackageReleaseNotes>
  </PropertyGroup>
```

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (reyang) [inline src/OpenTelemetry/OpenTelemetry.csproj:30]: I think some transformation is needed to convert relative links to absolute ones before publishing. Could you check the package and see if things are working as expected?

**Comment by reyang** [comment]
> Fixes #4895 Design discussion issue #4895

I don't think the PR fixes #4895 completely, there are several NuGet packages produced by this repository, and we need to cover all of them. I suggest that @RichardChukwu we change the wording from "fixes" to "related" so the issue won't be automatically closed, then address the other packages in follow up PRs once this PR got merged.

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (CodeBlanch) [reply src/OpenTelemetry/OpenTelemetry.csproj:32]: Should we move this to the props file so it applies to all packages?

In here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/19e1663c1e9485948d522d4f6d77ab652d6033bc/build/Common.prod.props#L13

Thinking something like...

```diff
    <PackageThirdPartyNoticesFile>$(RepoRoot)\THIRD-PARTY-NOTICES.TXT</PackageThirdPartyNoticesFile>
+   <PackageReadmeFile>README.md</PackageReadmeFile>
+   <PackageReleaseNotes>$(RepoRoot)\RELEASENOTES.md</PackageReleaseNotes>
  </PropertyGroup>
```

**Comment by RichardChukwu** [comment]
> I don't think the PR fixes #4895 completely, there are several NuGet packages produced by this repository, and we need to cover all of them. I suggest that @RichardChukwu we change the wording from "fixes" to "related" so the issue won't be automatically closed, then address the other packages in follow up PRs once this PR got merged.

Done @reyang thank you for the feedback

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (RichardChukwu) [reply src/OpenTelemetry/OpenTelemetry.csproj]: @reyang I automated the Link Transformation process during the build process so that before the NuGet package is published, relative links are converted to absolute links.

did this by updating the .csproj file to include a custom MSBuild task that processes the README.md file, replacing relative links with absolute ones. 

Do have a look please

Comment (reyang) [reply src/OpenTelemetry/OpenTelemetry.csproj]: Have you tested the output and what are the results? The change doesn't seem correct to me. Consider marking the PR as draft, validate locally (e.g. nuget files are zip files that you can unzip and examine the content), when things look correct to you, mark the PR as ready for review.

Comment (RichardChukwu) [reply src/OpenTelemetry/OpenTelemetry.csproj]: Hello @reyang thanks alot for the feedback. I edited the cs.proj transformation task and unziped the nuget file to confirm the README.md links were transformed. Kindly verify and give feedback please.

**Comment by RichardChukwu** [comment]
@reyang friendly ping please for another review on this PR

**Review by CodeBlanch — CHANGES_REQUESTED** [review]
I converted this PR to a draft because it seems to be modifying the main README, clobbering the SDK README, and adding a duplicate RELEASENOTES. The link transformation also doesn't seem robust.

A better design would be:

* Add a link to SDK CHANGELONG in SDK README only.
* Transform all links in a generic way during pack. Not sure if this is necessary, by the way. Does NuGet handle this? Please confirm first.

I would also like to see a change made which would work for all packages (see: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5885/commits/48c86d9b18d84253c9a13a95cf0a9b0ca4671028#r1788279565) but I'll leave that up to you @RichardChukwu so long as the above criteria is met.

**Comment by RichardChukwu** [comment]
Thanks for the feedback @CodeBlanch 

I'll look into your proposed design.  For transforming all links during pack, from my test, the NuGet package seems to handle that actually from my observation.

I'm new here though, so not sure exactly and open to correction.

File: README.md
Comment (RichardChukwu) [inline README.md:1]: Hello @CodeBlanch for this, I'm not sure how to make a change which would work for all packages as suggested, I'll appreciate further clarification on that please

Comment (CodeBlanch) [reply README.md:1]: I pushed changes to your branch to get you started. Please do me a favor and run a dotnet pack and then verify the packages created for the repo contain README.md (specific to each project) and RELEASENOTES.md (shared by all). They will have `.nupkg` extensions, just rename them `.zip` to open. Relative links should also be converted to GitHub permalinks automatically during pack.

Comment (RichardChukwu) [reply README.md:1]: Alright, thank you. I'll do that and give feedback soon

Comment (RichardChukwu) [reply README.md:1]: Hi @CodeBlanch dotnet pack ran as suggested locally after the changes and the link transformation worked 

Kindly let me know if there are further changes that need to be made. Thank you

**Comment by RichardChukwu** [comment]
Still awaiting your feedback please @CodeBlanch

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (CodeBlanch) [inline src/OpenTelemetry/OpenTelemetry.csproj:7]: @RichardChukwu Please stop pushing changes to this file. Or any other files 🤣 All of this is going to be handled via the props. I'm working to validate it works when rendered on NuGet.

Comment (RichardChukwu) [reply src/OpenTelemetry/OpenTelemetry.csproj]: ooh, I'm sorry about that, didn't know @CodeBlanch

**Comment by CodeBlanch** [comment]
I've been testing this via a repo I set up: https://github.com/CodeBlanch/NuGetReadMeTest

Here's an example of a pushed package:

https://www.nuget.org/packages/NuGetReadMeTest.ClassLibrary/1.0.0.3

Good news:

* Got the automatic relative link conversion working.

Bad news:

* The "Release Notes" tab is more limited than the "README" tab. You can't give it a file. And it doesn't support markdown if you give it a string. What I decided to do instead of stuffing the RELEASENOTES.md file into there (which would look terrible) I just linked to it. And also included a link to CHANGELOG.md which will be the one for the project/package.

**Comment by RichardChukwu** [comment]
@CodeBlanch 
Thank you for the update, and for testing everything out so thoroughly! It's great to hear that the automatic relative link conversion is working.

I understand the limitations with the "Release Notes" tab, and your solution of linking to the RELEASENOTES.md and CHANGELOG.md files makes sense to avoid formatting issues.

Given these changes, would you say the solution is now ready for merge, or are there any final adjustments you'd like me to make before proceeding?

Thanks again for your guidance!

**Review by CodeBlanch — APPROVED** [review]
LGTM

There are some CI failures but they don't seem related. Going to merge and look at those separately.

**Pull request merged**
